vivarium_public_health/risks/distributions.py

```diff
@@ -384,9 +385,46 @@ def ppf(self, x: pd.Series, interpolation: bool = True) -> Union[np.ndarray, pd.
                        for name, dist in self._distributions.items()], axis=0)
 
 
-def get_distribution(risk: str, risk_type: str, builder) -> Union[BaseDistribution, EnsembleDistribution]:
+class CategoricalDistribution:
+    def __init__(self, exposure_data: pd.DataFrame, risk: str):
+        self.exposure_data = exposure_data
+        self._risk = risk
+
+    def setup(self, builder):
+        self.exposure = builder.value.register_value_producer(f'{self._risk}.exposure',
+                                                              source=builder.lookup.build_table(self.exposure_data))
+    @staticmethod
+    def _get_sorted_categories(data):
+        import re
+        new_keys = lambda key: [int(elem) if elem.isdigit() else elem.lower() for elem in re.split('([0-9]+)', key)]
```

Comment:
Named lambda's almost always mean you're doing something too complicated for a lambda function.  This seems like the case here.